src/listen/ldap_sync/proto_ldap_sync_ldap.c

```diff
@@ -0,0 +1,1325 @@
+
+static CONF_PARSER sasl_mech_static[] = {
+	{ FR_CONF_OFFSET("mech", FR_TYPE_STRING | FR_TYPE_NOT_EMPTY, proto_ldap_sync_ldap_t, handle_config.admin_sasl.mech) },
+
+	{ FR_CONF_OFFSET("proxy", FR_TYPE_STRING, proto_ldap_sync_ldap_t, handle_config.admin_sasl.proxy) },
+
+	{ FR_CONF_OFFSET("realm", FR_TYPE_STRING, proto_ldap_sync_ldap_t, handle_config.admin_sasl.realm) },
+
+	CONF_PARSER_TERMINATOR
+};
+
+static CONF_PARSER ldap_sync_search_config[] = {
+	{ FR_CONF_OFFSET("base_dn", FR_TYPE_STRING, sync_config_t, base_dn), .dflt = "", .quote = T_SINGLE_QUOTED_STRING },
+
+	{ FR_CONF_OFFSET("filter", FR_TYPE_STRING, sync_config_t, filter) },
+
+	{ FR_CONF_OFFSET("scope", FR_TYPE_STRING, sync_config_t, scope_str), .dflt = "sub" },
+	/* For persistent search directories, setting this to "no" will load the whole directory. */
+	{ FR_CONF_OFFSET("changes_only", FR_TYPE_BOOL, sync_config_t, changes_only), .dflt = "yes" },
+
+	CONF_PARSER_TERMINATOR
+};
+
+static CONF_PARSER option_config[] = {
+	{ FR_CONF_OFFSET("idle", FR_TYPE_TIME_DELTA, proto_ldap_sync_ldap_t, handle_config.keepalive_idle), .dflt = "60" },
+
+	{ FR_CONF_OFFSET("probes", FR_TYPE_UINT32, proto_ldap_sync_ldap_t, handle_config.keepalive_probes), .dflt = "3" },
+
+	{ FR_CONF_OFFSET("interval", FR_TYPE_TIME_DELTA, proto_ldap_sync_ldap_t, handle_config.keepalive_interval), .dflt = "30" },
+
+	{ FR_CONF_OFFSET("dereference", FR_TYPE_STRING, proto_ldap_sync_ldap_t, handle_config.dereference_str) },
+	/* allow server unlimited time for search (server-side limit) */
+	{ FR_CONF_OFFSET("srv_timelimit", FR_TYPE_TIME_DELTA, proto_ldap_sync_ldap_t, handle_config.srv_timelimit), .dflt = "60" },
+	/* timeout for results */
+	{ FR_CONF_OFFSET("res_timeout", FR_TYPE_TIME_DELTA, proto_ldap_sync_ldap_t, handle_config.res_timeout), .dflt = "20" },
+	/* timeout on network activity */
+	{ FR_CONF_OFFSET("net_timeout", FR_TYPE_TIME_DELTA, proto_ldap_sync_ldap_t, handle_config.net_timeout), .dflt = "10" },
+	/* timeout for reconnections */
+	{ FR_CONF_OFFSET("reconnection_delay", FR_TYPE_TIME_DELTA, proto_ldap_sync_ldap_t, handle_config.reconnection_delay), .dflt = "5" },
+
+	CONF_PARSER_TERMINATOR
+};
+
+static CONF_PARSER tls_config[] = {
+	{ FR_CONF_OFFSET("ca_file", FR_TYPE_FILE_INPUT, proto_ldap_sync_ldap_t, handle_config.tls_ca_file) },
+	{ FR_CONF_OFFSET("ca_path", FR_TYPE_FILE_INPUT, proto_ldap_sync_ldap_t, handle_config.tls_ca_path) },
+	{ FR_CONF_OFFSET("certificate_file", FR_TYPE_FILE_INPUT, proto_ldap_sync_ldap_t, handle_config.tls_certificate_file) },
+	{ FR_CONF_OFFSET("private_key_file", FR_TYPE_FILE_INPUT, proto_ldap_sync_ldap_t, handle_config.tls_private_key_file) },
+	{ FR_CONF_OFFSET("start_tls", FR_TYPE_BOOL, proto_ldap_sync_ldap_t, handle_config.start_tls), .dflt = "no" },
+	{ FR_CONF_OFFSET("require_cert", FR_TYPE_STRING, proto_ldap_sync_ldap_t, handle_config.tls_require_cert_str) },
+
+	CONF_PARSER_TERMINATOR
+};
+
+static const CONF_PARSER global_config[] = {
```

Comment:
These are global

Reply:
Not really sure the offsets here are right.  It's probably only working because `handle_config` appears first in `proto_ldap_sync_ldap_t`?

Reply:
The elements it refers to are in `proto_ldap_sync_ldap_t` - it's the same as in rlm_ldap where the elements exist in `rlm_ldap_t`.  They are not in `handle_config`